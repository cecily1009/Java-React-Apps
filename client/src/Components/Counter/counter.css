.counter-body {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    text-align: center;
    margin-top: 20px;
}

.btn-group {
    display: grid;
    grid-template-columns: 100px 100px 100px;
    grid-template-rows: 80px 80px 80px 80px;
    gap: 15px 10px;
}

.reset-btn {
    width: 320px;
    height: 80px;
}

.number {
    margin-top: 20px;
    text-align: center;
    font-size: 5.5rem;
    font-weight: 800;
}