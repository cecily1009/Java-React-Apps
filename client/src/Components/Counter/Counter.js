import React,{useState} from 'react'
import './counter.css';
export const Counter = () => {
    const [number, setNumber] = useState(0);
    
    return (
        
        <div>
            <h1>Counter</h1>
            <div className="btn-group">
                <button className="btn btn-success" onClick={()=>setNumber(number+1)}>+1</button>
                <button className="btn btn-success" onClick={()=>setNumber(number+2)}>+2</button>
                <button className="btn btn-success" onClick={()=>setNumber(number+3)}>+3</button>
                <button className="btn btn-success" onClick={()=>setNumber(number+4)}>+4</button>
                <button className="btn btn-success" onClick={()=>setNumber(number+5)}>+5</button>
                <button className="btn btn-success" onClick={()=>setNumber(number+6)}>+6</button>
                <button className="btn btn-success" onClick={()=>setNumber(number+7)}>+7</button>
                <button className="btn btn-success" onClick={()=>setNumber(number+8)}>+8</button>
                <button className="btn btn-success" onClick={()=>setNumber(number+9)}>+9</button>
                <button className="btn btn-success" onClick={()=>setNumber(number+10)}>+10</button>
                <button className="btn btn-success" onClick={()=>setNumber(number-1)}>-1</button>
                <button className="btn btn-success" onClick={()=>setNumber(number-10)}>-10</button>
            </div>
            <div></div>
            <h1 className="number">{number}</h1>
            <button className="btn btn-danger reset-btn" onClick={()=>setNumber(0)}>Reset</button>
        </div>
    )
}
