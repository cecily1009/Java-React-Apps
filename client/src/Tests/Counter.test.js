import {shallow} from 'enzyme';
import Counter from '../Components/Counter/Counter';

// test('renders learn react link', () => {
//   render(<App />);
//   const h1Element = screen.getByText(/TodoList App/i);
//   expect(h1Element).toBeInTheDocument();
// });
describe('Counter', ()=>{
  it('should render correctly with no props',()=>{
    const component = shallow(<Counter />);
    expect(component).toMatchSnapshot();

  });

})
