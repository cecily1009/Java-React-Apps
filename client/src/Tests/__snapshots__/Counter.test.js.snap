// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter should render correctly with no props 1`] = `ShallowWrapper {}`;
